### PR TITLE
[Merged by Bors] - feat(NumberTheory/Padics): p-adic valuation of casts

### PR DESCRIPTION
**Pull request by loefflerd**
Lemmas about `Padic.valuation` applied to casts from various other rings to Qp

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/NumberTheory/Padics/PadicNumbers.lean [hunk 1]
```diff
@@ -969,27 +959,39 @@ theorem norm_eq_pow_val {x : ℚ_[p]} : x ≠ 0 → ‖x‖ = (p : ℝ) ^ (-x.va
...
+lemma valuation_intCast (n : ℤ) : valuation (n : ℚ_[p]) = padicValInt p n := by
+  rw [← Rat.cast_intCast, valuation_ratCast, padicValRat.of_int]
+
+@[simp]
+lemma valuation_natCast (n : ℕ) : valuation (n : ℚ_[p]) = padicValNat p n := by
+  rw [← Rat.cast_natCast, valuation_ratCast, padicValRat.of_nat]
+
+@[simp]
+lemma valuation_one : valuation (1 : ℚ_[p]) = 0 := by
+  rw [← Nat.cast_one, valuation_natCast, padicValNat.one, cast_zero]
```
Comment (Ruben-VandeVelde) [inline Mathlib/NumberTheory/Padics/PadicNumbers.lean:980]: Please also add `valuation_ofNat` with `OfNat.ofNat` and use it for `valuation_one`. (Don't forget to cargo-cult the `no_index` stuff.)

**Review by Ruben-VandeVelde — APPROVED** [review]

File: Mathlib/NumberTheory/Padics/PadicNumbers.lean (hunk 1 above)
Comment (loefflerd) [reply Mathlib/NumberTheory/Padics/PadicNumbers.lean:980]: What is this about `no_index`? I've not heard of it before.

**Comment by loefflerd** [comment]
OK, done. I don't claim to understand what this `no_index` stuff is about, but I followed some other examples in the library. It doesn't work to derive `valuation_one` from this, since it raises an error without `n.AtLeastTwo` in the statement of the `OfNat` lemma.

**Comment by MichaelStollBayreuth** [comment]
LGTM.
maintainer merge

**Comment by ocfnash** [comment]
Thanks!

bors merge

**Pull request closed without merging**